src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -341,4 +342,45 @@ public static void validateArraySchema(Schema arraySchema, String name) {
                                                        name, componentSchema));
     }
   }
+
+  public static void validateColumnForPartition(String columnName, Schema outputSchema) {
+    if (columnName == null) {
+      return;
+    }
+    if (outputSchema == null) {
+      throw new InvalidStageException(String.format("Can't validate column '%s' because output schema not specified",
+                                                       columnName));
+    }
+    for (Schema.Field field : outputSchema.getFields()) {
+      if (field.getName().equals(columnName)) {
+        if (!field.getSchema().getType().equals(Schema.Type.UNION)) {
+          if (field.getSchema().getLogicalType() != null
+            && (field.getSchema().getLogicalType().equals(Schema.LogicalType.DATE)
+            || field.getSchema().getLogicalType().equals(Schema.LogicalType.TIMESTAMP_MICROS))) {
```

Comment:
Even though most plugins don't use it, the sink should also allow TIMESTAMP_MILLIS in case somebody has written their own plugin that uses it.

Reply:
Done.